ddcommon-ffi/src/vec.rs

```diff
@@ -73,7 +74,7 @@ impl<'a, T> IntoIterator for &'a Vec<T> {
     type IntoIter = core::slice::Iter<'a, T>;
 
     fn into_iter(self) -> Self::IntoIter {
-        unsafe { self.as_slice().into_slice() }.iter()
+        self.deref().iter()
```

Comment:
Would `*self` work here instead?

Reply:
Yes, it would be `(*self).iter()` but that felt uncomfortable so I just used `.deref()` instead. However, the compiler sometimes auto-derefs so this could possibly just be `self.iter()`, but that feels a bit more magical and I'm not sure code style wise what would be preferred.

Reply:
I have seen clippy (in other situations) suggesting to remove `.deref()` or the `*` because it would auto-deref anyway, but I personally prefer the explicit deref over the implicit one.